CoreScriptsRoot/Modules/BackpackScript3D.lua

```diff
@@ -153,6 +179,8 @@ local function OnCharacterAdded(character)
 	
 	backpack.ChildAdded:connect(OnChildAdded)
 	backpack.ChildRemoved:connect(OnChildRemoved)
+
+	humanoid.Changed:connect(function() UpdateHealth(humanoid) end)
```

Comment:
We should call update health here too, you might be given a half damaged character.